### PR TITLE
refactor(frontend): merge sorting util with pinning utils for tokens

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

To reduce the computational time, we merge two of the loops needed for the derived store `combinedDerivedSortedNetworkTokensUi`:
1. First loop to create props `balance` and `usdBalance`, done in store `combinedDerivedEnabledNetworkTokensUi`.
2. Second loop to separate the tokens with balance, from the one that do not have it.

# Changes

- Add the `balance` and `usdBalance` calculation in function `pinTokensWithBalanceAtTop`.
- Adapt tests to the new function, providing a simpler mocked `usdValue` function.
- Adapt derived store `combinedDerivedSortedNetworkTokensUi` to feed the correct store stores the new function.
- Removed deprecated unused store `combinedDerivedEnabledNetworkTokensUi`.
- Unrelated: we renamed a few mock values for module `tokens.utils.spec`, to make the code less verbose.

# Tests

No changes in local replica.

Screenshot 2024-09-02 at 13 45 34

### CODE REVIEW

File: src/frontend/src/lib/utils/tokens.utils.ts [hunk 1]
```diff
@@ -55,17 +59,45 @@ export const sortTokens = ({
...
+					})
+				: undefined;
+
+			const tokenUI: TokenUi = {
+				...token,
+				balance,
+				usdBalance
+			};
+
+			(usdBalance ?? 0) > 0 || (balance ?? ZERO).gt(0)
```
Comment (peterpeterparker) [inline src/frontend/src/lib/utils/tokens.utils.ts:95]: Not necessarely for this PR and a bit an feedback but, given that the function isn't a single line anymore, do we have chance here to introduce a `if` and transform the `push` into a more functional programming approach?

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks

File: src/frontend/src/lib/utils/tokens.utils.ts (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/utils/tokens.utils.ts]: i will do it in this one

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/utils/tokens.utils.ts]: @peterpeterparker before I merge, have a look, if that what you meant

Comment (peterpeterparker) [reply src/frontend/src/lib/utils/tokens.utils.ts]: Almost there, still not fully functional. We can keep the push in this PR and try to do that later or not.

**Pull request merged**